File: manager/src/main/java/moe/shizuku/manager/home/ManageAppsViewHolder.java
package moe.shizuku.manager.home;

import android.content.Context;
import android.content.Intent;
import android.view.View;

import moe.shizuku.api.ShizukuService;
import moe.shizuku.manager.management.ApplicationManagementActivity;
import moe.shizuku.manager.R;
import rikka.html.widget.HtmlCompatTextView;
import rikka.recyclerview.BaseViewHolder;

public class ManageAppsViewHolder extends BaseViewHolder<Integer> implements View.OnClickListener {

    public static final Creator<Integer> CREATOR = (inflater, parent) -> new ManageAppsViewHolder(inflater.inflate(R.layout.item_home_manage_apps, parent, false));

    private HtmlCompatTextView title;
    private HtmlCompatTextView summary;

    public ManageAppsViewHolder(View itemView) {
        super(itemView);

        title = itemView.findViewById(android.R.id.text1);
        summary = itemView.findViewById(android.R.id.text2);

        itemView.setOnClickListener(this);
    }

    @Override
    public void onBind() {
        Context context = itemView.getContext();

        title.setHtmlText(context.getResources().getQuantityString(R.plurals.authorized_apps_count, getData(), getData()));

        if (!ShizukuService.pingBinder()) {
            itemView.setEnabled(false);
            summary.setHtmlText(context.getString(R.string.service_not_running, context.getString(R.string.service_name)));
        } else {
            itemView.setEnabled(true);
            summary.setHtmlText(context.getString(R.string.view_authorized_apps));
        }
    }

    @Override
    public void onClick(View v) {
        v.getContext().startActivity(new Intent(v.getContext(), ApplicationManagementActivity.class));
    }
}

File: manager/src/main/java/moe/shizuku/manager/home/ManageAppsViewHolder.kt
package moe.shizuku.manager.home

import android.content.Intent
import android.text.method.LinkMovementMethod
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import moe.shizuku.manager.Helps
import moe.shizuku.manager.R
import moe.shizuku.manager.databinding.HomeItemContainerBinding
import moe.shizuku.manager.databinding.HomeManageAppsItemBinding
import moe.shizuku.manager.ktx.toHtml
import moe.shizuku.manager.management.ApplicationManagementActivity
import moe.shizuku.manager.model.ServiceStatus
import rikka.html.text.HtmlCompat
import rikka.recyclerview.BaseViewHolder
import rikka.recyclerview.BaseViewHolder.Creator

class ManageAppsViewHolder(private val binding: HomeManageAppsItemBinding, root: View) :
    BaseViewHolder<Pair<ServiceStatus, Int>>(root), View.OnClickListener {

    companion object {
        val CREATOR = Creator<Pair<ServiceStatus, Int>> { inflater: LayoutInflater, parent: ViewGroup? ->
            val outer = HomeItemContainerBinding.inflate(inflater, parent, false)
            val inner = HomeManageAppsItemBinding.inflate(inflater, outer.root, true)
            ManageAppsViewHolder(inner, outer.root)
        }
    }

    init {
        root.setOnClickListener(this)
    }

    private inline val title get() = binding.text1
    private inline val summary get() = binding.text2

    override fun onBind() {
        val context = itemView.context
        if (!data.first.isRunning) {
            itemView.isEnabled = false
            title.setText(R.string.home_app_management_title)
            summary.text = context.getString(
                R.string.home_status_service_not_running,
                context.getString(R.string.app_name)
            )
        } else {
            itemView.isEnabled = true
            title.text = context.resources.getQuantityString(
                R.plurals.home_app_management_authorized_apps_count,
                data.second,
                data.second
            )
            summary.text = context.getString(R.string.home_app_management_view_authorized_apps)
        }
    }

    override fun onClick(v: View) {
        v.context.startActivity(Intent(v.context, ApplicationManagementActivity::class.java))
    }
}
